From 5345af3dd7e4dcb37d52f0c12e4b0c8dc7a6c888 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 15:30:08 -0400
Subject: [block] mtip32xx: Remove superfluous call to pci_disable_msi()

Message-id: <1402673421-17117-6-git-send-email-snitzer@redhat.com>
Patchwork-id: 83921
O-Subject: [RHEL7.1 PATCH 05/18] mtip32xx: Remove superfluous call to pci_disable_msi()
Bugzilla: 1105703
RH-Acked-by: David Milburn <dmilburn@redhat.com>

BZ: 1105703

Upstream commit cf91f39b1704f456b00ddaa1a7294919d97fb355
Author: Alexander Gordeev <agordeev@redhat.com>
Date:   Wed Feb 19 09:58:15 2014 +0100

    mtip32xx: Remove superfluous call to pci_disable_msi()

    There is no need to call pci_disable_msi() in case
    the previous call to pci_enable_msi() failed

    Signed-off-by: Alexander Gordeev <agordeev@redhat.com>
    Cc: Jens Axboe <axboe@kernel.dk>
    Cc: Asai Thambi S P <asamymuthupa@micron.com>
    Cc: linux-pci@vger.kernel.org
    Signed-off-by: Jens Axboe <axboe@fb.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/block/mtip32xx/mtip32xx.c b/drivers/block/mtip32xx/mtip32xx.c
index ad26ddd..32e4aa3 100644
--- a/drivers/block/mtip32xx/mtip32xx.c
+++ b/drivers/block/mtip32xx/mtip32xx.c
@@ -4623,7 +4623,7 @@ static int mtip_pci_probe(struct pci_dev *pdev,
  if (rv) {
   dev_warn(&pdev->dev,
    "Unable to enable MSI interrupt.\n");
-  goto block_initialize_err;
+  goto msi_initialize_err;
  }
 
  /* Initialize the block layer. */
@@ -4653,6 +4653,8 @@ static int mtip_pci_probe(struct pci_dev *pdev,
 
 block_initialize_err:
  pci_disable_msi(pdev);
+
+msi_initialize_err:
  if (dd->isr_workq) {
   flush_workqueue(dd->isr_workq);
   destroy_workqueue(dd->isr_workq);
-- 
1.7.1